### PR TITLE
[v5] System for all actor logic

### PR DESCRIPTION
**Pull request by davidkpiano**
This PR provides access to `system` for all actor logic.

```ts
fromPromise(({ system }) => { ... });

fromTransition((state, event, { system }) => { ... });

fromObservable(({ system }) => { ... });

fromEventObservable(({ system }) => { ... });

fromCallback((sendBack, receive, { system }) => { ... });
```

### CODE REVIEW

File: packages/core/test/system.test.ts [hunk 1]
```diff
@@ -269,4 +274,53 @@ describe('system', () => {
...
+        }
+      ]
+    });
+
+    const actor = interpret(machine).start();
+
+    expect(actor.system.get('test')).toBeDefined();
+
+    // The assertion won't be checked until the transition function gets an event
+    actor.system.get('reducer')!.send({ type: 'a' });
```
Comment (Andarist) [inline packages/core/test/system.test.ts:324]: why is that? 🤔

File: packages/core/test/system.test.ts [hunk 2]
```diff
@@ -269,4 +274,53 @@ describe('system', () => {
 
     interpret(machine).start();
   });
+
+  it('should be accessible in other actors', () => {
+    expect.assertions(6);
```
Comment (Andarist) [inline packages/core/test/system.test.ts:279]: could u split this test into 6 separate tests?

**Review by mellson — APPROVED** [review]
I guess my comment is just a type-level nit, so there is no reason why you shouldn't get an approval

File: packages/core/src/actors/observable.ts [hunk 3]
```diff
@@ -131,7 +146,12 @@ export function fromObservable<T, TEvent extends EventObject>(
  */
 
 export function fromEventObservable<T extends EventObject>(
-  lazyObservable: ({ input }: { input: any }) => Subscribable<T>
+  lazyObservable: ({
+    input
```
[the remark this answers is not in the document]
Comment (davidkpiano) [reply packages/core/src/actors/observable.ts:150]: Will do this in a follow-up PR, good idea

**Pull request merged**